components/ai_chat/core/browser/conversation_driver.cc

```diff
@@ -642,6 +656,42 @@ bool ConversationDriver::IsPageContentsTruncated() {
           GetCurrentModel().max_page_content_length);
 }
 
+bool ConversationDriver::HasPendingConversationEntry() {
+  return pending_conversation_entry_ != nullptr;
+}
+
+bool ConversationDriver::IsContentAssociationPossible() {
+  const GURL url = GetPageURL();
+
+  if (!base::Contains(kAllowedSchemes, url.scheme())) {
+    return false;
+  }
+
+  return true;
+}
+
+void ConversationDriver::SubmitSummarizationRequest() {
+  DCHECK(IsContentAssociationPossible())
+      << "This conversation request is not associated with content\n";
+
+  mojom::ConversationTurn turn = {
+      CharacterType::HUMAN, ConversationTurnVisibility::VISIBLE,
+      l10n_util::GetStringUTF8(IDS_CHAT_UI_SUMMARIZE_PAGE)};
+  MakeAPIRequestWithConversationHistoryUpdate(std::move(turn));
+  pending_message_needs_page_content_ = true;
```

Comment:
so, should `MakeAPIRequestWithConversationHistoryUpdate` support a flag that would be checked in the `if` statement you commented on below? this is specifically for users who haven't opted in, as we halt messages while they're signing the agreement. tt seems like a specialized case, warranting a separate function with an appropriate name. 

i'm struggling to see how this would make for a better API. perhaps there's a use case i'm not aware of. 

i went ahead and made the necessary changes in this commit: